zebra-chain/src/orchard/note/nullifiers.rs

```diff
@@ -0,0 +1,84 @@
+#![allow(clippy::unit_arg)]
+#![allow(dead_code)]
+
+use halo2::{arithmetic::FieldExt, pasta::pallas};
+
+use crate::serialization::serde_helpers;
+
+use super::super::{
+    commitment::NoteCommitment, keys::NullifierDerivingKey, note::Note, sinsemilla::*,
+};
+
+/// A cryptographic permutation, defined in [poseidonhash].
+///
+/// PoseidonHash(x, y) = f([x, y, 0])_1 (using 1-based indexing).
+///
+/// [poseidonhash]: https://zips.z.cash/protocol/nu5.pdf#poseidonhash
+fn poseidon_hash(_x: pallas::Base, _y: pallas::Base) -> pallas::Base {
+    unimplemented!()
+}
+
+/// Used as part of deriving the _nullifier_ for a Orchard _note_.
+///
+/// PRF^nfOrchard: F_𝑞P × F_𝑞P → F_𝑞P
+///
+/// Instantiated using the PoseidonHash hash function defined in [§5.4.1.10
+/// ‘PoseidonHash Function’][poseidon]:
+///
+/// PRF^nfOrchard(nk*, ρ*) := PoseidonHash(nk*, ρ*)
+///
+/// [concreteprfs]: https://zips.z.cash/protocol/protocol.pdf#concreteprfs
+/// [poseidonhash]: https://zips.z.cash/protocol/nu5.pdf#poseidonhash
+fn prf_nf(nk: pallas::Base, rho: pallas::Base) -> pallas::Base {
+    poseidon_hash(nk, rho)
+}
+
+/// A Nullifier for Orchard transactions
+#[derive(
+    Clone,
+    Copy,
+    Debug,
+    Eq, // Hash,
```

Comment:
What does `// Hash,` mean here?

Reply:
Hash is a std Rust trait, I think it requires Copy

Reply:
But why is it commented out?

Reply:
Removed with a2cfa295, it was there as `sapling::Nullifier`, a newtype around bytes, had it, but `pallas::Base`, does not derive it. If we need it, we may need to replace the inner type with bytes.